app/admin/notifications.rb

```diff
@@ -0,0 +1,24 @@
+ActiveAdmin.register Notification do
+  permit_params :user, :details, :viewed
+
+  index sortable: :viewed_asc do
+    column 'Sender', sortable: :user_id do |notification|
+      link_to(notification.user.email, admin_user_path(notification.user)) + "  (#{notification.user.to_full_name})"
+    end
+    column :details
+    column :viewed
+    actions
+  end
+
+  form do |f|
+    f.inputs 'Notifications' do
+      f.input :user, collection: User.all.map { |u| ["#{u.email} (#{u.first_name} #{u.last_name})", u.id] }
```

Comment:
@SerhiiBorozenets you can use model method `.to_full_name` but it's better to rename this method before to `full_name`

f.input :user, collection: User.all.map { |u| ["#{u.email} (#{u.full_name})", u.id] }